### ISSUE TITLE
Can't link SDK Plugin when not using an Xcode Workspace

### ISSUE THREAD

**Issue by IVeselovskyy** [mopub/mopub-unity-sdk#51]
We build iOS project with the following parameters for the resolver:
![ResolverSettings](https://user-images.githubusercontent.com/23023940/55427166-6a448500-5554-11e9-9821-2d135771be8c.png)

Unity SDK doesn't get linked properly when using the local pods and adding them to the Xcode project. I've managed to avoid similar issues for the adapters (mentioned in #50) by using the version hosted on Cocoapods instead of the local one, but Unity SDK is missing from there, so there's no way to do it.  

Although it seems to be working when switching to workspace option, our build scripts rely on having everything in one project. So unfortunately just switching to using workspace is quite problematic. 
- [*] I am submitting a bug report for existing functionality
- [*] I visited https://developers.mopub.com/ and found no answer
- [*] I checked https://twittercommunity.com/c/advertiser-api/mopub to make sure that this issue has not already been filed
- [*] I checked to make sure that this issue has not already been filed

#### MoPub SDK Version: 
5.6.0

#### Device model and OS Version:
iOS

#### Unity Version: 
2017.4.16f1

#### Evidence:
Podfile:
```
source 'https://github.com/CocoaPods/Specs.git'
install! 'cocoapods', :integrate_targets => false
platform :ios, '9.0'

target 'Unity-iPhone' do
pod 'MoPub-AdMob-Adapters', '7.39.0.0', :subspecs => ['Network']
pod 'MoPub-FacebookAudienceNetwork-Adapters', '5.1.0.4', :subspecs => ['Network']
pod 'mopub-ios-sdk', '5.6.0', :subspecs => ['Core']
pod 'MoPub-IronSource-Adapters', '6.8.1.0.1', :subspecs => ['Network']
pod 'MoPub-SDK-Plugin', :path => '/Users/***/Assets/MoPub/Plugins/iOS'
pod 'MoPub-TapJoy-Adapters', '12.2.0.4', :subspecs => ['Network']
pod 'MoPub-UnityAds-Adapters', '3.0.0.4', :subspecs => ['Network']
pod 'MoPub-Vungle-Adapters', '6.3.2.4', :subspecs => ['Network']
end
```

Xcode fails to link the library:
```
ld: directory not found for option '-L /Users/****/Library/Developer/Xcode/DerivedData/Unity-iPhone-****/Build/Intermediates.noindex/ArchiveIntermediates/Unity-iPhone/BuildProductsPath/Release-iphoneos/MoPub-SDK-Plugin
ld: library not found for -lMoPub-SDK-Plugin
```
It's also missing from the list in the Xcode:
![image](https://user-images.githubusercontent.com/23023940/55427790-c1972500-5555-11e9-8863-ca1f6e3d0884.png)

**Comment by MixMasterRickyC** [comment]
We had the same issue, in the end we removed the line 

`<iosPod name="MoPub-SDK-Plugin" bitcodeEnabled="false" minTargetSdk="9.0" path="Assets/MoPub/Plugins/iOS"/>`

from `Assets\MoPub\Scripts\Editor\MoPubDependencies.xml`

Since before the framework used to live in that folder.

Also make sure the binding scripts in `Assets/MoPub/Plugins/iOS` have their import settings set to iOS, since after updating our seems to be set to none.

That seem to get it compiling for us.
Hope this helps.

**Comment by IVeselovskyy** [comment]
@MixMasterRickyC Thanks! It did make it build, but it looks like it's not working even with the guaranteed ad units :(

**Comment by sdolim** [comment]
@MixMasterRickyC That's the correct fix for the time being to get it to build.  It appears that the Unity Jar Resolver doesn't quite handle "local pods" (the ones with path= instead of version= attributes in the XML file) when the Cocoapods Integration mode is configured for "Xcode project" instead of "Xcode workspace".  

@IVeselovskyy Can you provide details on your current status (error messages, etc)?

**Comment by IVeselovskyy** [comment]
@sdolim 
We were getting an error 
`Adapter MPMoPubRewardedVideoCustomEvent failed to load ad: (32) Failed to load creative:
[Various metadata regarding the ad]`

Mediated ad providers seemed to have been working, just not the Marketplace. 
We ended up modifying our build scripts to work with the workspace solution, and it seems that now the issue is gone. I think the issue was with linking the mopub-ios-sdk.

**Comment by sdolim** [comment]
@IVeselovskyy I think I found the problem now:  There are several resources in mopub-ios-sdk/MoPubSDK/ that are not being copied into the app.  These are: Internal/Common/MPAdBrowserController.xib and all the files in Resources/.  

I was able to fix this by adding these in the Build Phases > Copy Files step.  I've attached a screenshot of what this looks like when done.  Be sure to untick the "Code Sign on Copy" checkbox next to MRAID.bundle too.

<img width="734" alt="Screen Shot 2019-04-03 at 12 35 30 PM" src="https://user-images.githubusercontent.com/5303868/55507780-4abd6300-560d-11e9-9420-0adb53b5b2be.png">

(This agains seems to be an issue with how the Unity Jar Resolver is handling a local pod in the "Xcode project" mode of cocoapods integration.)

**Comment by IVeselovskyy** [comment]
Great, thanks!

**Comment by sdolim** [comment]
@IVeselovskyy I'm working with the Unity Jar Resolver project to find a "permanent" fix for your issue, and I may have one but I'm wondering if you could test if it will work with your build system.  Here are the steps of the experiment:

1.  Set the Jar Resolver back to "Xcode Workspace" integration.
2.  Delete any previous Xcode project folder that Unity generated last time you did Build (or Build & Run).
3.  Build again in Unity to have it generate the Xcode project again.
4.  (Optional) In the Xcode project folder, you can delete the Unity-iPhone.xcworkspace (you won't be needing it).
5.  Open the Unity-iPhone Xcode project in Xcode.
6.  Go to File > Add Files, and in the dialog, select the Pods/Pods.xcproject to get it added to the Unity-iPhone project as a subproject.
7.  Add a Target Dependency to the Unity-iPhone target and select Pods-Unity-iPhone (from the Pods subproject).
8.  At this point the project is ready to build the app, either in Xcode (Cmd-B build) or on the command line ("xcodebuild -project Unity-iPhone.xcodeproj -target Unity-iPhone").

Here's a pic of what I'm saying:

<img width="960" alt="Screen Shot 2019-04-16 at 1 03 28 PM" src="https://user-images.githubusercontent.com/5303868/56241536-f4a7e100-604a-11e9-878d-ce79f4abd905.png">

My question is:  Will the Xcode project as amended by the above steps work in your build system?  This is using the older subproject method that the workspace method superceded, but it is still available and seems to work fine for Unity iOS app builds. 

If this does work, my "permanent" fix will be to automate those steps so they happen during Unity's Build phase, and no manual intervention will be needed.

**Comment by sdolim** [comment]
Obviously my Step 1 and Step 2 in the picture don't line up with the steps in my text (should be Step 6 and Step 7)...

**Comment by sdolim** [comment]
Interestingly, I have tried leaving out Step 7 (adding the target dependency - Step 2 in the pic), and the project still seems to build the stuff under Pods first.  I'm not sure if that's a fluke or if it is being smart.

**Comment by sdolim** [comment]
I submitted a patch to the Jar Resolver maintainers that fixes this problem.  We'll include a new release of the resolver in our next SDK release.

**Comment by aornelas** [comment]
thanks for reporting!

this has been fixed in the [5.7.1 release](https://github.com/mopub/mopub-unity-sdk/releases/tag/v5.7.1)
